fenix/app/src/main/java/org/mozilla/fenix/compose/SwipeToDismiss.kt

```diff
@@ -0,0 +1,153 @@
+    dismissThreshold: ThresholdConfig = FractionalThreshold(DISMISS_THRESHOLD),
+    backgroundContent: @Composable RowScope.() -> Unit,
+    dismissContent: @Composable RowScope.() -> Unit,
+) {
+    val swipeWidth = with(LocalDensity.current) {
+        LocalConfiguration.current.screenWidthDp.dp.toPx()
+    }
+    val anchors = mutableMapOf(0f to Default)
+    val thresholds = { _: DismissValue, _: DismissValue ->
+        dismissThreshold
+    }
+
+    if (StartToEnd in directions) anchors += swipeWidth to DismissedToEnd
+    if (EndToStart in directions) anchors += -swipeWidth to DismissedToStart
+
+    Box(
+        Modifier
+            .swipeable(
+                state = state,
+                anchors = anchors,
+                thresholds = thresholds,
+                orientation = Orientation.Horizontal,
+                enabled = state.currentValue == Default && enabled,
+                reverseDirection = LocalLayoutDirection.current == LayoutDirection.Rtl,
+                resistance = null,
+            )
+            .then(modifier),
+    ) {
+        Row(
+            content = backgroundContent,
+            modifier = Modifier.matchParentSize(),
+        )
+
+        Row(
+            content = dismissContent,
+            modifier = Modifier.offset { IntOffset(state.offset.value.roundToInt(), 0) },
+        )
+    }
+}
+
+private const val DISMISS_THRESHOLD = 0.5f
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun SwipeablePreview(directions: Set<DismissDirection>, text: String) {
+    val state = rememberDismissState()
+
+    FirefoxTheme {
+        Box(
+            modifier = Modifier
+                .height(30.dp)
+                .fillMaxWidth(),
+        ) {
+            SwipeToDismiss(
+                state = state,
+                directions = directions,
+                backgroundContent = {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .background(FirefoxTheme.colors.layerAccent),
+                    )
+                },
+            ) {
+                Row(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .background(FirefoxTheme.colors.layer1),
+                    horizontalArrangement = Arrangement.Center,
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    Text(text)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SwipeToDismissPreview() {
+    Column {
+        SwipeablePreview(directions = setOf(StartToEnd), text = "Swipe to right ->")
+        Spacer(Modifier.height(30.dp))
+        SwipeablePreview(directions = setOf(EndToStart), text = "<- Swipe to left")
+        Spacer(Modifier.height(30.dp))
+        SwipeablePreview(directions = setOf(StartToEnd, EndToStart), text = "<- Swipe both ways ->")
+    }
```

Comment:
love how succinct this preview is to show how to use SwipeToDismiss! `dismissThreshold` is the only non-obvious parameter not captured you might want to add an example for.